src/transactions/feed/queryHelper.ts

```diff
@@ -210,16 +263,67 @@ export function useFetchTransactions(): QueryHookResult {
   }
 }
 
-async function queryTransactionsFeed(
-  address: string | null,
-  localCurrencyCode: string,
+function anyChainHasMorePages(pageInfo: { [key in Chain]?: PageInfo | null }): boolean {
+  return Object.values(pageInfo).reduce((acc, val) => acc || !!val?.hasNextPage, false)
+}
+
+function anyChainHasTransactionsOnCurrentPage(hasTransactionsOnCurrentPage: {
+  [key in Chain]?: boolean
+}): boolean {
+  return Object.values(hasTransactionsOnCurrentPage).reduce((acc, val) => acc || val, false)
+}
+
+// Queries for transactions feed for any number of chains in parallel,
+// with optional pagination support.
+async function queryTransactionsFeed({
+  address,
+  localCurrencyCode,
+  params,
+}: {
+  address: string | null
+  localCurrencyCode: string
+  params: Array<{
+    chain: Chain
+    afterCursor?: string
+  }>
+}): Promise<{ [key in Chain]?: QueryResponse }> {
+  const results = await Promise.all(
+    params.map((params) =>
```

Comment:
the callback's "params" argument should not shadow the variable name from outer scope, very confusing